### DISCUSSION TITLE
Unable to load object variant by field

### DISCUSSION THREAD

**Discussion by TS1997** [pimcore/pimcore#9762] [Q&A / Support] [answered]
Hello!

I'm currently developing a site for a customer and am currently trying to use the generated Class::getByFieldX functions to load products. It works as expected and I get the correct object when it's an object but whenever I try to load a variant of that object (which has a unique value, i.e. not an inherited value) I get null as a response (I have a limit of 1).
I'm running the latest Pimcore 10.0.7. 

Is this a bug or am I missing something?

**Comment by dvesh3** [comment, marked as answer]
@TS1997 On 10.1 version, this will be supported with `objectTypes` array param
e.g. `ClassName::getByMyfieldname($value, ['limit' => $limit, 'offset' => $offset, 'objectTypes' => [DataObject::OBJECT_TYPE_VARIANT]]);`

please see implementation here https://github.com/pimcore/pimcore/pull/9634

And with Pimcore 11, the behavior will be changed to return variants by default https://github.com/pimcore/pimcore/issues/9686

**Reply by TS1997** [reply]
@dvesh3 Thanks for the details! Is there a set release date for 10.1? I'm in need of this ASAP

**Reply by dvesh3** [reply]
you check due date on milestones https://github.com/pimcore/pimcore/milestone/121
